pom.xml

```diff
@@ -70,6 +77,12 @@
 			<artifactId>mapstruct</artifactId>
 			<version>${mapstruct.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>javax.annotation</groupId>
+			<artifactId>javax.annotation-api</artifactId>
```

Comment:
Na moją aktualną wiedzę to Spring Boot 3 korzysta z Jakarty, a nie z Javax.

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0.0-M1-Release-Notes#minimum-requirements-changes

Jakarta EE 9

Migracja Spring Boot v2.7 → v3.0
https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee

As well as dependency coordinate changes, Jakarta EE now uses jakarta packages rather than javax.

Ja dałbym tą zależność
https://mvnrepository.com/artifact/jakarta.annotation/jakarta.annotation-api/3.0.0-M1

Reply:
Podbicie wersji hibernate-jpamodelgen pozwoliło uniknąc korzystania z javax, przez to powyższa dependency w pom.xml nie jest już potrzebna